### ISSUE TITLE
CoffeeScript by CoffeeScripters show. 

### ISSUE THREAD

**Issue by skw** [nodeup/contribute#2]
### [NodeUp #60](http://nodeup.com/)

With:
- @dshaw
- @brettkiefer
- @mcculloughsean
- @davidguttman 
- @spikebrehm  

Topics:
- Best-practices to have CoffeeScript in node modules?
- Thoughts on node.js + CoffeeScript, does it have a place?
- Is there a workflow for some JS/CS devs working on the same project?
- Compile to JS before publishing to npm.
  - https://groups.google.com/forum/#!msg/npm-/fm83SDRrpI8/SlxABF_Uf8EJ

**Comment by dshaw** [comment]
Thanks for creating the issue for this!

**Comment by mhart** [comment]
Perhaps it could be a general "transpiling" show chatting about compiling into JS from any lang, ie try and get some TypeScript people in, code contract libraries, JSX, hell even browserify transpiles.

**Comment by skw** [comment]
Forgive my ignorance, but I was under the impression that typescript never really caught on, I haven't seen it in any node module sources. I would gladly be corrected.

**Comment by dshaw** [comment]
@mhart Good idea, but probably a separate show.

**Comment by mhart** [comment]
@skw it's still under active development, cf: http://blogs.msdn.com/b/typescript/archive/2013/06/18/announcing-typescript-0-9.aspx

**Comment by paulbjensen** [comment]
I couldn't imagine an episode about CoffeeScript being done on NodeUp. Pleasantly surprised. As a fan of metaphors, anyone who has watched the tv show "The Bridge" will be interested to note that Danes and Swedes speak to each other in their own languages, and they understand each other. I hoped that the relationship between JS/CS would be like that, but I'm doubtful.

**Comment by dshaw** [comment]
@paulbjensen Did you catch the Promises show? I think that was very constructive and good for everyone. This is about giving CoffeeScript developers their voice and helping them connect within the Node community. I actually probably won't have any of the "Node ppls" on, with the exception of myself for that reason.

**Comment by simianhacker** [comment]
Talk to the We Are Fractal guys (Github: wearefractal, contra, funkytek) They can provide some good perspectives on why they choose CoffeeScript over JavaScript

**Comment by paulbjensen** [comment]
@dshaw I haven't, but I will when I get a moment. I'm a big fan of CoffeeScript, wrote quite a few apps in it, including Dashku.

**Comment by skw** [comment]
I still think it would be really cool if you could get Jeremy and Trevor on the show ;D

**Comment by TrevorBurnham** [comment]
@skw Aw, that's very kind of you to say, but trust me: It would be 1000% better if you had just Jeremy on the show. And perhaps Michael Ficarra of [CoffeeScript Redux](https://github.com/michaelficarra/CoffeeScriptRedux) fame. I was fortunate enough to help bring CoffeeScript to a wide audience in its early days, but I don't have much to say about it now that it's a fairly well-known coding tool.

**Comment by dshaw** [comment]
@aseemk has an interesting CoffeeScript preso. Might be a good fit for this show. http://aseemk.com/talks/intro-to-coffeescript

I especially appreciate his perspective as a previous CS doubter who embraced CS fully. #fullkoolaid

(Yes, for those playing along at home, I do disagree with most of the arguments of why CS is great in that preso. But this is about you... not me. :smile:)

**Comment by Siyfion** [comment]
Personally, I'd really like to see this happen! As a developer choosing between "native" JS and others (CoffeeScript, TypeScript, Dart) for both front-end and back-end development, I'd really love to hear a constructive argument (keyword: **constructive**) for all sides involved.

**Comment by dshaw** [comment]
We touched on this in the NodeUp Teams show with Team Trello. Worth checking out when it goes up.

**Comment by Siyfion** [comment]
Ah thanks @dshaw, I heard in the last podcast that you were discussing getting the Trello team involved, as a user of Trello, I have a lot of respect for what they've done; glad to hear that there is some discussion on this. Any idea when it'll be up?

**Comment by dshaw** [comment]
@BrettKiefer has very much the right attitude for this show. We might invite him back again to reprise that discussion. I also like that his background was JavaScript (and before that Haskell), so he also fails the "you like CoffeeScript because you like Ruby/Python" assumption.

**Comment by dshaw** [comment]
@Siyfion We had some technical issues because we had too many people on, so I think @floating is going to have some extra work this time to clean things up. Probably by Friday, though.

**Comment by Siyfion** [comment]
Awesome, I'll look forward to that then! Keep up the good work. :+1:

**Comment by optikfluffel** [comment]
+1

**Comment by dshaw** [comment]
@mcculloughsean Would you like to be on this show?

**Comment by mcculloughsean** [comment]
@dshaw Sure. I'd highly recommend getting @michaelficarra involved too.

**Comment by dshaw** [comment]
@mcculloughsean Sounds good.

**Comment by uris77** [comment]
+1 for this show.

**Comment by aseemk** [comment]
Sorry I never responded to my mention above: Thank you for the consideration. I'd be down!

**Comment by tj** [comment]
whyyyy am i getting emails :(

**Comment by tj** [comment]
@skw could you take me out of the desc? It won't let me unsub

**Comment by dshaw** [comment]
@visionmedia Click the unsubscribe link on this issue? Haven't used that yet. (You know you want to be on this show! :smile: )

**Comment by tj** [comment]
it keeps emailing me anyway haha, I have my settings set to email on mentions but I guess each comment here is a mention as well

**Comment by dshaw** [comment]
Sucks. Sorry about that.

**Comment by dshaw** [comment]
@visionmedia The original issue had you mentioned in it. I removed that mention. Hopefully that clears things up. :dancer:

**Comment by tj** [comment]
I think that did it! thanks

**Comment by jxson** [comment]
I highly recommend @davidguttman

**Comment by michaelficarra** [comment]
If you're looking for CoffeeScript maintainers, @satyr, @Nami-Doc, and I would all be fine choices. Though @satyr is Japanese and @Nami-Doc is French, so I can't vouch for their spoken English abilities.

**Comment by davidguttman** [comment]
Sure, I'm down. Sounds like fun. Thanks @jxson =)

**Comment by vendethiel** [comment]
I'm not that confident into my accent but at dotRB I've been able to speak and understand with english-talking people without much trouble, so I don't know (what do you mean "it's not hard to be better than most french ?". I'm kidding, of course -- ha ! I know we're bad at english. I'm much more familiar with american accents - I learned english mostly with RailsCasts). I certainly search words sometimes, though.
I'm not an experienced speaker at all (I did say I want to fix that before I'm 20, and being in Paris makes it easy to participate in meetups) so I'm probably not game for a 1h+ podcast "alone" but I'd love being here with @michaelficarra , @satyr -- or anybody else interested, really.

TL;DR: seems AWESOME & scary

**Comment by paulbjensen** [comment]
I'd totally be up for hearing from these peeps, I love CoffeeScript and I'll probably name my first born Jeremy in honour of the creator.

**Comment by emkay** [comment]
+1 for @davidguttman

**Comment by dshaw** [comment]
Paging @spikebrehm with the contrarian view of a CS codebase working back to JS.

**Comment by spikebrehm** [comment]
I'd love to contribute with lessons from our experience at Airbnb working with a large JavaScript/CoffeeScript codebase with ~40 engineers contributing.  We introduced CS in 2011 and, being a Rails shop, we loved it, mostly for the `class` keyword and fat arrow.  But, we've since stopped writing CS, preferring idiomatic JavaScript.  Our switch was due to a few big reasons:
- CoffeeScript is optimized for writing, not reading.  We found it hard to grok the dense, terse CS written by other engineers, and there are so many ways to write CS.
- Line numbers. Once we started working in Node.js, the mismatched line numbers on exception stack traces were maddening, especially during the heat of the moment of an outage.
- Open source. Once we started open-sourcing some of our CS code, we saw that it became a barrier to adoption and contribution of the community.

**Comment by vendethiel** [comment]
> Line numbers. Once we started working in Node.js, the mismatched line numbers on exception stack traces were maddening, especially during the heat of the moment of an outage.

That should be better now, though.

> Open source. Once we started open-sourcing some of our CS code, we saw that it became a barrier to adoption and contribution of the community.

That might be true, but that's also an anti-argument because it creates a vicious circle. If you want people to partake in a community, you need to be the first to do so. I understand, though, obviously.

**Comment by uris77** [comment]
Why can't it be that people like both? Liking and using CS shouldn't mean you don't use or dislike JS, and vice-versa. There seems to be an almost religious/fanatical following or 'perception' that does more harm than good. I personally use both and love both. Using different languages doesn't equate to cheating on your spouse. It seems to me that sometimes we tend to conflate them.

**Comment by dshaw** [comment]
This is why we can't have nice things. :cake:

**Comment by dshaw** [comment]
:coffee: :ship:!

**Comment by vendethiel** [comment]
Well then, looking forward to it :).

**Comment by AsaAyers** [comment]
I didn't follow exactly who made the suggestion, but I opened an issue for [Catch comprehensions without parens](https://github.com/clutchski/coffeelint/issues/237)

**Comment by brettkiefer** [comment]
@AsaAyers I was the one bemoaning this, and thank you for opening the issue! How . . . constructive.
